src/backlight/backlight.c

```diff
@@ -378,26 +397,33 @@ int main(int argc, char *argv[]) {
 
         if (streq(argv[1], "load")) {
                 _cleanup_free_ char *value = NULL;
-                const char *clamp;
+                bool clamp;
 
                 if (shall_restore_state() == 0)
                         return EXIT_SUCCESS;
 
                 if (!validate_device(udev, device))
                         return EXIT_SUCCESS;
 
-                r = read_one_line_file(saved, &value);
-                if (r < 0) {
+                clamp = shall_clamp(device);
 
+                r = read_one_line_file(saved, &value);
+                if (r == -ENOENT && clamp) {
+                        /* not saved: fake "0" to trigger brightness clamping if necessary */
+                        value = strdup("0");
+                        if (!value) {
+                                log_oom();
+                                return EXIT_FAILURE;
+                        }
+                } else if (r < 0) {
                         if (r == -ENOENT)
                                 return EXIT_SUCCESS;
```

Comment:
hmm, so we check r == -ENOENT here twice now. Please rework this as:

```c
r = read_one_line_file(saved, &value);
if (r == -ENOENT) {
        if (!clamp)
            return EXIT_SUCCESS;
        value = strdup(…
        …
} else if (r < 0) {
        log_error_errno(…
…
```

Reply:
Done.